### ISSUE TITLE
runtime error: invalid memory address or nil pointer 

### ISSUE THREAD

**Issue by jishnurajendran** [ssleert/zfxtop#9]
`panic: runtime error: invalid memory address or nil pointer dereference `
` [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x55be51826947] `

After running for a 1 or 2 seconds this error shows up.
I installed zfxtop from AUR

**Comment by ssleert** [comment]
hmm, is there more information?

**Comment by jishnurajendran** [comment]
The error is printed over the zfxtop outputs it looks a bit strange

```

   │ Free 44%           6.62 GiB │ │ ■■■■■■■■■■■■■■■■■■■■■■■panic: runtime error: invalid memory address or nil pointer dereference───────────────────────╯
   │                             │ ╭─  IN[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4ab2c7]  ▆▆▆▆▆▆ │ │  Distro  EndeavourOS Linux │
   │ ▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁██████ │ │  Hostname        qphy-qphy │
   ╰────────────────────goroutine 35 [running]:──────────────────╯
                         Q - quit | R - redraw github.com/ssleert/zfxtop/internal/data.getDiskInfo.func3.1({0xc000310d60?, 0x20?}, {0x0, 0x0}, {0x100?, 0xffffffffffffff9c?})
                                                                                    	/home/ssleert/User/git/zfxtop/internal/data/depent.go:469 +0x27
                                                              path/filepath.walk({0xc0001cf860, 0x17}, {0x4fe248, 0xc0002b35f0}, 0xc00020ffa0)
                                                     	/usr/lib/go/src/path/filepath/path.go:500 +0x1f3
               path/filepath.walk({0xc00001bb60, 0xf}, {0x4fe248, 0xc0002b3520}, 0xc00020ffa0)
     	/usr/lib/go/src/path/filepath/path.go:504 +0x26f
                                                        path/filepath.walk({0xc0001b99a0, 0xa}, {0x4fe248, 0xc000417380}, 0xc00020ffa0)
                                              	/usr/lib/go/src/path/filepath/path.go:504 +0x26f
       path/filepath.walk({0x4d86a7, 0x4}, {0x4fe248, 0xc000222000}, 0xc00020ffa0)
                                                                                  	/usr/lib/go/src/path/filepath/path.go:504 +0x26f
                                               path/filepath.Walk({0x4d86a7, 0x4}, 0xc0000d17a0)
       	/usr/lib/go/src/path/filepath/path.go:571 +0x6c
                                                       github.com/ssleert/zfxtop/internal/data.getDiskInfo.func3(0x0?, 0x0?)
                                   	/home/ssleert/User/git/zfxtop/internal/data/depent.go:467 +0x105
               created by github.com/ssleert/zfxtop/internal/data.getDiskInfo
                                                                             	/home/ssleert/User/git/zfxtop/internal/data/depent.go:456 +0x205
```
     
**Kernel: 6.1.12-arch1-1 
Shell: fish 3.6.0 
Terminal: kitty**

**Comment by ssleert** [comment]
Please try the build last commit
If the problem goes away I will create a release

**Comment by jishnurajendran** [comment]
Its not fixed yet, tried AUR,  building and installing from script. 
I have AMD CPU, and I use asusctl for battery management.

**Comment by ghost** [comment]
Yep, I have a similar error.

`   │ Free 64%           9.82 GiB │ │ ■■■■■■■■■■■■■■■■■■■■■■panic: runtime error: invalid memory address or nil pointer dereference
   │                             │ ╰─────────────────────────────╯                                                                [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4aaf87]
   │                      ▄▄▄▄▄▄ │ ╭─  INFO ────────────────────╮
   │                      ██████ │ │  Distro         Arch Linux │                                                                                                                                       goroutine 34 [running]:
   │ ▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁██████ │ │  Hostname            hello │                                                                                                                                                              github.com/ssleert/zfxtop/internal/data.getDiskInfo.func2.1({0xc0003b02e0?, 0x1d?}, {0x0, 0x0}, {0x100?, 0xffffffffffffff9c?})
                         Q - quit | R - redraw                                                                          	/home/ssleert/User/git/zfxtop/internal/data/depent.go:486 +0x27
                                                                                                                                                                                               path/filepath.walk({0xc000527968, 0x16}, {0x4ff028, 0xc0000b2f70}, 0xc000293fa0)
                                         	/usr/lib/go/src/path/filepath/path.go:500 +0x1f3
                                                                                                path/filepath.walk({0xc000294390, 0x11}, {0x4ff028, 0xc000223790}, 0xc000293fa0)
                                                                                                                                                                                	/usr/lib/go/src/path/filepath/path.go:504 +0x26f
  path/filepath.walk({0xc000394020, 0x9}, {0x4ff028, 0xc0003960d0}, 0xc000293fa0)
                                                                                 	/usr/lib/go/src/path/filepath/path.go:504 +0x26f
                                                                                                                                        path/filepath.walk({0x4d97f5, 0x5}, {0x4ff028, 0xc000396000}, 0xc000293fa0)
                                                                                                                                                                                                                   	/usr/lib/go/src/path/filepath/path.go:504 +0x26f
                                  path/filepath.Walk({0x4d97f5, 0x5}, 0xc000136fa0)
                                                                                   	/usr/lib/go/src/path/filepath/path.go:571 +0x6c
                                                                                                                                       github.com/ssleert/zfxtop/internal/data.getDiskInfo.func2(0x0?, 0x0?)
                                                                                                                                                                                                            	/home/ssleert/User/git/zfxtop/internal/data/depent.go:484 +0x105
                                          created by github.com/ssleert/zfxtop/internal/data.getDiskInfo
                                                                                                        	/home/ssleert/User/git/zfxtop/internal/data/depent.go:473 +0x195`

But it may be because my processor is x86_64 bit, however, a few weeks ago I didn't have this problem.

By the way: it would be cool to add an option in the configuration file to also display the remaining battery time.

**Comment by ssleert** [comment]
fixed in 0.3.2 version
